modules/core/arrow-core/gradle.properties

```diff
@@ -1,4 +1,4 @@
 # Maven publishing configuration
 POM_NAME=Arrow Core Extensions
-POM_ARTIFACT_ID=arrow-core-extensions
```

Comment:
@raulraja @JorgeCastilloPrz didn't we agree that there'd be arrow-core-data, arrow-core-extensions; and a separate package arrow-core that'd only reexport both?

Reply:
That changed, because of unnecessary overhead. 
Now there is: 
- arrow-core-data
- arrow-core 

Same in mtl:
- arrow-mtl-data
- arrow-mtl

@raulraja @JorgeCastilloPrz should we do the same for free, effects. and other modules ?